[18.0][MIG] spreadsheet_oca: Migration to 18.0

chrisandrewmann opened the pull request:
Continues and supercedes odoowired's PR https://github.com/OCA/spreadsheet/pull/65
I'll summarise below.

### Forward ported PRs from V17
- https://github.com/OCA/spreadsheet/pull/69
- https://github.com/OCA/spreadsheet/pull/70
- https://github.com/OCA/spreadsheet/pull/71

### Fixes done over original PR
1. Crashed when refreshing/browsing the URL of a spreadsheet action record in browser due to V18 routing changes
2. bus_service connection changes to enable the undo/redo and collaborative editing (thanks to @etobella)
3. Global filters were not applying and are now fixed
4. Pivot data source menus crashed
5. Chart data import crashed when editing the chart properties
6. Fix listview import button not appearing in all cases (when Upload button present for example Vendor Bills)
7. Add logic for handling spreadsheet messages for SNAPSHOT and CLIENT events
8. Restructured to use new SpreadsheetComponent class, which solves the issue of ConfirmationDialogs using browser JS window alerts
9. [IMP] Make all new pivots dynamic by default and format as tables
10. [IMP] Add support for import images feature
11. [IMP] Add accesskey shortcuts (ALT+S) for all spreadsheet import buttons
12. [IMP] Add feature to re-import lists from data sources in side-panel with selectable number of rows

etobella committed:
[ADD] spreadsheet_oca

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[IMP] spreadsheet_oca: Allow to import to existent spreadsheets

etobella committed:
[IMP] spreadsheet_oca: Allow to export graphs to spreadsheets

etobella committed:
[FIX] spreadsheet_oca: Improve security

With this change we are adding an extra layer of security. Without it, any user was able to sniff how all happened using something like:

    const any_spreadsheet_id = 1234;
    const channel = "spreadsheet_oca;spreadsheet.spreadsheet;" + any_spreadsheet_id;
    bus_service.addChannel(channel);
    bus_service.addEventListener("spreadsheet_oca", (message) => /* every revision arrives here */

With the change, we verify the access to the model with a similar logic to `web_editor` fields

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
spreadsheet_oca 16.0.1.0.1

etobella committed:
[IMP] spreadsheet_oca: add to current sheet

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
spreadsheet_oca 16.0.1.1.0

etobella committed:
[FIX] spreadsheet_oca: Ctrl+Z was not working properly

etobella committed:
[IMP] spreadhseet_oca: Allow edition of sheet name

etobella committed:
[IMP] spreadsheet_oca: Allow to edit filters

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
spreadsheet_oca 16.0.1.1.1

Commit:
[UPD] Update spreadsheet_oca.pot

etobella committed:
[FIX] spreadsheet_oca: Filters where failing

etobella committed:
[IMP] spreadsheet_oca: Alloe to edit Pivot data

etobella committed:
[IMP] spreadsheet_oca: Allow to import lists

etobella committed:
[IMP] spreadsheet_oca: Allow to edit lists

etobella committed:
[IMP] spreadsheet_oca: Allow to reinsert a pivot table

etobella committed:
[FIX] spreadsheet_oca: Fix filters

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
spreadsheet_oca 16.0.1.1.2

etobella committed:
[IMP] spreadsheet_oca: Allow to edit Odoo graphs properly

Fixes #9

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
spreadsheet_oca 16.0.1.1.3

Commit:
[UPD] Update spreadsheet_oca.pot

etobella committed:
[IMP] spreadsheet_oca: Add roadmap for specific questions

etobella committed:
[IMP] spreadsheet_oca: Allow to set the menu in all kind of charts. Improve change of kind

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
spreadsheet_oca 16.0.1.2.0

OCA-git-bot committed:
[UPD] README.rst

Jan-Onestein committed:
Added translation using Weblate (Dutch)

Jan-Onestein committed:
Translated using Weblate (Dutch)

Currently translated at 83.1% (69 of 83 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/nl/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (83 of 83 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/it/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (83 of 83 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/it/

royle-vietnam committed:
[FIX] spreadsheet_oca: Cannot open spreadsheet when using undo / redo
commands

Steps:
1. Open spreadsheet
2. Press command Ctrl + Z
3. F5 (refresh browse)

An error occurred:
Operation undefined not found

Because missing key `undoneRevisionId` / `redoneRevisionId` in revisions
data

Solutions:
Save all spreadsheet message in commands field

legalsylvain committed:
[IMP] spreadsheet_oca : Improve documentation

Commit:
[UPD] Update spreadsheet_oca.pot

legalsylvain committed:
Added translation using Weblate (French (France) (fr_FR))

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

rbellanova committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (83 of 83 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/it/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

daiduongnguyen-odoo committed:
[FIX] spreadsheet_oca: error when adding relation filter

Close https://github.com/OCA/spreadsheet/issues/24 and
https://github.com/OCA/spreadsheet/issues/27

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

chrisandrewmann committed:
[FIX] spreadsheet_oca: Error when deleting a worksheet
Closes https://github.com/OCA/spreadsheet/issues/32

OCA-git-bot committed:
[BOT] post-merge updates

CarlosRoca13 committed:
[FIX] spreadsheet_oca: Pivot does not imported correctly

Steps to reproduce the problem:

1. Go to a pivot
2. Group by something on the cols or on the rows
3. Create a spreadsheet with that info

The pivot is imported on the initial state

CarlosRoca13 committed:
[IMP] spreadsheet_oca: Auto resize cols to adjust them to cotent

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (85 of 85 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/it/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

CarlosRoca13 committed:
[FIX] spreadsheet_oca: Keep pivot order

When adding a pivot table to a spreadsheet, it
does not maintain the specified order because the
sortedColumn option is not being completed.

Steps to reproduce the issue:

1. Go to a pivot table
2. Change the default order
3. Add to the spreadsheet

The added table will not have preserved the order.

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

CarlosRoca13 committed:
[IMP] spreadsheet*_oca: Add field to define name of table and restructure order of fields

CarlosRoca13 committed:
[IMP] spreadsheet_oca: Allow generate dynamic tables

On dashboards exists tables that are updated dynamically,
with this changes we have the option to generate this
dynamic tables using the pivot inserter.

Added a button to duplicate and change to dynamic
table the selected table on spreadsheet editor.

If a table with multiple row goups generated, the
way to generate the dynamic table will be set the
number of rows for each level of indentation.

Example:
2 rowGroupBys
3 number of rows
- val1
    - subval1.1
    - subval1.2
    - subval1.3
- val2
    - subval2.1
    - subval2.2
    - subval2.3
- val3
    - subval3.1
    - subval3.2
    - subval3.3

CarlosRoca13 committed:
[IMP] spreadsheet_oca: Allow download spreadsheet as XLSX

CarlosRoca13 committed:
[FIX] spreadsheet_oca: Make last changes compatibles with tree and graph

Since this changes the field datasource_name does not work with
tree and graph. Now the datasource_name is set as data name for the
created data.

Furthermore, the capability to create dynamic tables has no sense on
tree and graph, so it has been limited for pivots.

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (95 of 95 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/it/

peterromao committed:
Added translation using Weblate (Portuguese)

peterromao committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (95 of 95 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/pt/

royle-vietnam committed:
[FIX] spreadsheet_oca: correct spelling errors and remove outdated
translation terms

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

CarlosRoca13 committed:
[IMP] spreadsheet_*_oca: Add possibility to make columns dynamic

With these changes, we have added the ability to make the columns
dynamic, just as is done with the rows.

To add this type of table, we will need to select the dynamic columns
option and choose the number of columns to insert. Keep in mind that if
multiple column levels are selected, the chosen number will apply to
each of these levels.

In addition to this, the wizard that opens with the spreadsheet button
has been updated to allow us to choose the type of dynamism we expect in
the inserted table.

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (99 of 99 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/it/

CarlosRoca13 committed:
[FIX] spreadsheet_oca: Show the "Add" button in invoice list

If any module males a primary template from the ones where the button is
added, the button wasn't be shown previously.

That's because the inheritance and load order works in OWL. The way to
fix this is to make this load order deterministic specifying to be
loaded after the corresponding template for being taken into account for
primary views. This also required to split the templates file into
several ones.

gregory-moka committed:
Added translation using Weblate (French)

gregory-moka committed:
Translated using Weblate (French)

Currently translated at 63.6% (63 of 99 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/fr/

CarlosRoca13 committed:
[FIX] spreadsheet_oca: Avoid to show dynamic cols when no cols selected

Steps to reproduce the error:

1. Go to a pivot
2. Remove all groups of columns to set only the measure
3. Add the pivot to a spreadsheet with dynamic columns and a number of columns

An error will be thrown

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (101 of 101 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/it/

CarlosRoca13 committed:
[FIX] spreadsheet_oca: Clean searchParams of problematic context keys

Before these changes, when a table or chart was added with an active
user filter, the user's selections were not respected, either in the
groupings or in the measures.

Thanks to these changes, we remove the conflicting keys from the context
passed through the searchParams, thus respecting the user's selection in
the table/chart itself.

Steps to reproduce the problem:

1. Go to a pivot or graph
2. Save a filter
3. Modify the groups or measures
4. Add to a spreadsheet

Current behavior: The table will not be added with the current selection
but it will be added with the filters groups and measures.

Expected behavior:  The table will be added with the current selection.

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (101 of 101 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/it/

etobella committed:
[IMP] spreadsheet_oca: Show data only on debug mode. Otherwise it is confusing for users

OCA-git-bot committed:
[BOT] post-merge updates

etobella committed:
[IMP] spreadsheet_oca: Allow to import XLSX Files

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (102 of 102 strings)

Translation: spreadsheet-16.0/spreadsheet-16.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-16-0/spreadsheet-16-0-spreadsheet_oca/it/

luisDIXMIT committed:
[IMP] spreadsheet_oca: pre-commit auto fixes

luisDIXMIT committed:
[MIG] spreadsheet_oca: Migration to 17.0

etobella committed:
[MIG] spreadsheet_oca: Continue migration

chrisandrewmann committed:
[IMP] spreadsheet_oca: Add support for changing locale and currencies.
Set default currency for new sheets.
Fix crash on insertDynamicPivot.
Fix typos (noficiation > notification).
Add feature to delete pivots and lists from sheet via button on sidebars.

etobella committed:
[FIX] spreadsheet_oca: Fix when using dynamic number of rows & columns

chrisandrewmann committed:
[FIX] Crashes downloading XLSX. Crashes adding chart to sheet.

chrisandrewmann committed:
[IMP+MIG] Merge PR (https://github.com/OCA/spreadsheet/pull/57) from 16.0.
Rename field data > spreadsheet_binary_data and add pre-migrate script (V17 changed)

chrisandrewmann committed:
[FIX] Dynamic rows and columns generation crashed, rewrite

chrisandrewmann committed:
[IMP+FIX] Add icons to file menu. Change date range filters to Month-Quarter/Relative/From-To (aligned with EE), Fix default value preference on global filter function.

chrisandrewmann committed:
[FIX] Remove coding pragma and add contributor

chrisandrewmann committed:
[IMP] spreadsheet_oca: pre-commit auto fixes

chrisandrewmann committed:
[IMP] Move spreadsheet data fields into spreadsheet_abstract and add init logic to set default locale of new sheets

Commit:
[UPD] Update spreadsheet_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

xtanuiha committed:
Added translation using Weblate (Chinese (Simplified) (zh_CN))

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (102 of 102 strings)

Translation: spreadsheet-17.0/spreadsheet-17.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-17-0/spreadsheet-17-0-spreadsheet_oca/it/

Commit:
[UPD] Update spreadsheet_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: spreadsheet-17.0/spreadsheet-17.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-17-0/spreadsheet-17-0-spreadsheet_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (119 of 119 strings)

Translation: spreadsheet-17.0/spreadsheet-17.0-spreadsheet_oca
Translate-URL: https://translation.odoo-community.org/projects/spreadsheet-17-0/spreadsheet-17-0-spreadsheet_oca/it/

chrisandrewmann committed:
[FIX] spreadsheet_oca: getGroupByDisplayLabel error

Fix bug where sort order in pivot causes crash due to changed function
name getGroupByDisplayLabel to getMeasureDisplayName.

OCA-git-bot committed:
[BOT] post-merge updates

CarlosRoca13 committed:
[FIX] spreadsheet_oca: Make use of _zip_xslx_files that spreadsheet module offers

Reduce the logic of the controller by using the function that spreadsheet module offers and avoid error when rendering images on the xlsx file.

OCA-git-bot committed:
[BOT] post-merge updates

CarlosRoca13 committed:
[FIX] spreadsheet_oca: Filters don't take care about field type

OCA-git-bot committed:
[BOT] post-merge updates

chrisandrewmann committed:
[IMP] spreadsheet_oca: pre-commit auto fixes

chrisandrewmann committed:
[MIG] spreadsheet_oca: Migration to 18.0

luisDIXMIT reviewed (commented):
At first, I was able to access and edit the spreadsheet, but after changing different things, I started getting this error when trying to edit a spreadsheet or create a new one. I can't tell you the steps to reproduce it, but I believe it crashed after I deleted a spreadsheet. I'm not completely sure.

<img width="1151" height="646" alt="image" src="https://github.com/user-attachments/assets/e377b420-861a-4d9b-a2b6-d3bf35cb5ca2" />

chrisandrewmann commented:
> At first, I was able to access and edit the spreadsheet, but after changing different things, I started getting this error when trying to edit a spreadsheet or create a new one. I can't tell you the steps to reproduce it, but I believe it crashed after I deleted a spreadsheet. I'm not completely sure.
> 
>

Hi @luisDIXMIT, interesting.
I can't reproduce it myself either, maybe you had multiple tabs opened and deleted the source spreadsheet in another?
Maybe able to put some better error handling in to deal with missing records. Can you try again to reproduce it so I can follow your steps?

luisDIXMIT commented:
> > At first, I was able to access and edit the spreadsheet, but after changing different things, I started getting this error when trying to edit a spreadsheet or create a new one. I can't tell you the steps to reproduce it, but I believe it crashed after I deleted a spreadsheet. I'm not completely sure.
> 
> Hi @luisDIXMIT, interesting. I can't reproduce it myself either, maybe you had multiple tabs opened and deleted the source spreadsheet in another? Maybe able to put some better error handling in to deal with missing records. Can you try again to reproduce it so I can follow your steps?

I only have one tab open. I'm going to reproduce it and provide you with more information.

luisDIXMIT commented:
@chrisandrewmann sorry mate i couldn't reproduce it :(

luisDIXMIT reviewed (approved):
Tested on runboat and LGTM!!

luisDIXMIT reviewed (commented):
A small comment: the paste function in edit entry doesn’t seem to be working. Otherwise, you’ve done a good job. Thanks.

KubeAl commented:
I have faced same issue in kubernetes. Happened once my pod killed and
restarted. It can be fixed by upgrading the module again.

On Thu, 2 Oct 2025 at 5:28 PM, Luis David Rodríguez <
***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> Tested on runboat and LGTM!!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/spreadsheet/pull/72#pullrequestreview-3294227282>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXAYNUHYRP2JEXTC3EGQ7CD3VUHOTAVCNFSM6AAAAACHMCULBSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTEOJUGIZDOMRYGI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

chrisandrewmann commented:
> A small comment: the paste function in edit entry doesn’t seem to be working. Otherwise, you’ve done a good job. Thanks.

Weird, I can't reproduce this on my prod system (inside latest official Odoo docker container) or via local Windows deployment.
I wonder if it's an infrastructure issue or maybe browser blocking?

UPDATE:
I've found the issue, it's because runboat is not SSL and browser security blocks clipboard and notifications by default.
If you test it on HTTPS connection you'll see it's fine.
<img width="1870" height="1229" alt="image" src="https://github.com/user-attachments/assets/425dc76c-cfb4-45af-ad1d-9593354a296e" />

chrisandrewmann commented:
@luisDIXMIT @KubeAl any thoughts or ideas around the issue I mentioned that I couldn't resolve?

_Collaborative editing no longer works like in V17. I've tried for days to solve it and given up, needs an OWL expert!
In V17 if multiple people edit the same spreadsheet, you should see the client username in a highlighted cell where they are currently editing the sheet. This doesn't work anymore._

KubeAl commented:
This one seems related to the asset handling. Regenerating assets fixes it.
Need to dwell on it more to understand this. Will update you.

On Thu, 2 Oct 2025 at 5:47 PM, Chris Mann ***@***.***> wrote:

> *chrisandrewmann* left a comment (OCA/spreadsheet#72)
> <https://github.com/OCA/spreadsheet/pull/72#issuecomment-3360930223>
>
> @luisDIXMIT <https://github.com/luisDIXMIT> @KubeAl
> <https://github.com/KubeAl> any thoughts or ideas around the issue I
> mentioned that I couldn't resolve?
>
>
> *Collaborative editing no longer works like in V17. I've tried for days to
> solve it and given up, needs an OWL expert! In V17 if multiple people edit
> the same spreadsheet, you should see the client username in a highlighted
> cell where they are currently editing the sheet. This doesn't work anymore.*
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/spreadsheet/pull/72#issuecomment-3360930223>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXAYNUCFK65MIXZ4UVWRD7L3VUJW7AVCNFSM6AAAAACHMCULBSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGNRQHEZTAMRSGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

chrisandrewmann commented:
> @chrisandrewmann sorry mate i couldn't reproduce it :(

I've managed to reproduce it now @luisDIXMIT! It was when Developer Mode (debug) was enabled, only then did it crash.
Just force-pushed an update, please retest.

- Fixed issue when developer mode was enabled, causing crash loading a spreadsheet
- Fixed crash after deleting a list data source, had to force reload of the sidepanel to prevent

CarlosRoca13 reviewed (changes requested):
The creation of dynamic rows/columns have been disappeared

luisDIXMIT commented:
> @luisDIXMIT @KubeAl any thoughts or ideas around the issue I mentioned that I couldn't resolve?
> 
> _Collaborative editing no longer works like in V17. I've tried for days to solve it and given up, needs an OWL expert! In V17 if multiple people edit the same spreadsheet, you should see the client username in a highlighted cell where they are currently editing the sheet. This doesn't work anymore._

Sorry mate, I'm not expert on OWL. Any other thing I will try to help you.

luisDIXMIT commented:
> > @chrisandrewmann sorry mate i couldn't reproduce it :(
> 
> I've managed to reproduce it now @luisDIXMIT! It was when Developer Mode (debug) was enabled, only then did it crash. Just force-pushed an update, please retest.
> 
> * Fixed issue when developer mode was enabled, causing crash loading a spreadsheet
> * Fixed crash after deleting a list data source, had to force reload of the sidepanel to prevent

Nice, I couldn't reproduce it so I think it's fixed, thanks!!

chrisandrewmann commented:
> The creation of dynamic rows/columns have been disappeared

Hi @CarlosRoca13
I can confirm that this is deliberate as i've used the principal to make new pivots dynamic by default so there is no need for setting the rows or columns when inserting a new pivot. In my opinion this is deprecated.
Then when inserting a new pivot, it is formatted automatically as a table and dynamic by default.
However I left the ability for you use the pivot sidebar buttons to insert a static pivot (Insert Pivot) or you can go to menu Data > Re-insert Static Pivot.

Note that nothing should have been changed in the list data source insert, I left that working as it was before.

CarlosRoca13 commented:
Hmmm, okay... But if I want, for example, that the pivot has 10 rows instead of 2 that are set on the pivot?

I don't see the way of doing this and I think it is a lost feature

chrisandrewmann commented:
> Hmmm, okay... But if I want, for example, that the pivot has 10 rows instead of 2 that are set on the pivot?
> 
> I don't see the way of doing this and I think it is a lost feature

I agree it is a compromise, however couldn't find an alternative and wanted to keep it simple.
If you want to make some code suggestions feel free but I'm struggling to spend any more time on it.

chrisandrewmann commented:
I made one last small update, added a feature which I feel is lacking and needed. Button on the list data side-panel.
I'm using it daily in production and one of my frustrations when a list changes length and was no way to reinsert it easily.
Hope that helps.

[IMP] Add feature to re-import lists from data sources in side-panel with selectable number of rows.

<img height="400" alt="image" src="https://github.com/user-attachments/assets/2d692554-6ff2-4b6b-85b2-3c996739a5d8" />

etobella commented:
Fixed the collaborative here: https://github.com/chrisandrewmann/spreadsheet/pull/1

chrisandrewmann commented:
> Fixed the collaborative here: [chrisandrewmann#1](https://github.com/chrisandrewmann/spreadsheet/pull/1)

Awesome! Thanks @etobella 
I was so nearly there, just missed that. Your expertise is much appreciated 😊

Merged and should be testable in here now.

pedrobaeza commented:
Please do a rebase a get rid-off the merge commit, and it would be nice if you squash administrative commits into the main one.

chrisandrewmann commented:
> Please do a rebase a get rid-off the merge commit, and it would be nice if you squash administrative commits into the main one.

Done!

etobella commented:
@CarlosRoca13 I tested locally and everything seemed to work. I see that you asked for some changed. They are still missing?

CarlosRoca13 commented:
> @CarlosRoca13 I tested locally and everything seemed to work. I see that you asked for some changed. They are still missing?

Hi @etobella The creation of dynamic pivots with different number of rows, but I can add it in other PR. So I will not block this more time

etobella reviewed (approved):
Ok! Let's proceed with merge so we can continue with this

/ocabot merge nobump

pedrobaeza commented:
Agh, I was reducing commit history and you let me no time...

pedrobaeza commented:
@CarlosRoca13 the dynamic thing is not needed anymore:

![Peek 09-10-2025 08-54](https://github.com/user-attachments/assets/256879b0-6a91-449f-be99-06c5c9a04fe4)

CarlosRoca13 commented:
> @CarlosRoca13 the dynamic thing is not needed anymore:
> 
> ![Peek 09-10-2025 08-54](https://private-user-images.githubusercontent.com/7165771/499205934-256879b0-6a91-449f-be99-06c5c9a04fe4.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk5OTMyNDMsIm5iZiI6MTc1OTk5Mjk0MywicGF0aCI6Ii83MTY1NzcxLzQ5OTIwNTkzNC0yNTY4NzliMC02YTkxLTQ0OWYtYmU5OS0wNmM1YzlhMDRmZTQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDlUMDY1NTQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmY4YjE1OTdmMTZjMTJjNDRhNmU0MWE1OGJkYmM1MGIyMTIyNjU3MmY1YWI2OWVlODVmOTM1NjFmN2E4YTEzNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.1PKJFYRwAw1RRlknHVEhgtdAed1eajid0m3Fv0uBLfQ) [ ![Peek 09-10-2025 08-54](https://private-user-images.githubusercontent.com/7165771/499205934-256879b0-6a91-449f-be99-06c5c9a04fe4.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk5OTMyNDMsIm5iZiI6MTc1OTk5Mjk0MywicGF0aCI6Ii83MTY1NzcxLzQ5OTIwNTkzNC0yNTY4NzliMC02YTkxLTQ0OWYtYmU5OS0wNmM1YzlhMDRmZTQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDlUMDY1NTQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmY4YjE1OTdmMTZjMTJjNDRhNmU0MWE1OGJkYmM1MGIyMTIyNjU3MmY1YWI2OWVlODVmOTM1NjFmN2E4YTEzNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.1PKJFYRwAw1RRlknHVEhgtdAed1eajid0m3Fv0uBLfQ) ](https://private-user-images.githubusercontent.com/7165771/499205934-256879b0-6a91-449f-be99-06c5c9a04fe4.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk5OTMyNDMsIm5iZiI6MTc1OTk5Mjk0MywicGF0aCI6Ii83MTY1NzcxLzQ5OTIwNTkzNC0yNTY4NzliMC02YTkxLTQ0OWYtYmU5OS0wNmM1YzlhMDRmZTQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDlUMDY1NTQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmY4YjE1OTdmMTZjMTJjNDRhNmU0MWE1OGJkYmM1MGIyMTIyNjU3MmY1YWI2OWVlODVmOTM1NjFmN2E4YTEzNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.1PKJFYRwAw1RRlknHVEhgtdAed1eajid0m3Fv0uBLfQ) [ ](https://private-user-images.githubusercontent.com/7165771/499205934-256879b0-6a91-449f-be99-06c5c9a04fe4.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk5OTMyNDMsIm5iZiI6MTc1OTk5Mjk0MywicGF0aCI6Ii83MTY1NzcxLzQ5OTIwNTkzNC0yNTY4NzliMC02YTkxLTQ0OWYtYmU5OS0wNmM1YzlhMDRmZTQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDlUMDY1NTQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmY4YjE1OTdmMTZjMTJjNDRhNmU0MWE1OGJkYmM1MGIyMTIyNjU3MmY1YWI2OWVlODVmOTM1NjFmN2E4YTEzNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.1PKJFYRwAw1RRlknHVEhgtdAed1eajid0m3Fv0uBLfQ)

I know that the pivots are dynamic by default, but in previous versions, I could add a dynamic pivot with more rows/columns than the shown on the pivot, now it is not possible. If you think that this is not necessary I will not invest time on recover it.

CarlosRoca13 commented:
Oh, okay, I see that the new row was added, sorry, I did not see it on my first watch

pedrobaeza commented:
/ocabot migration spreadsheet_oca
